packages/bp/src/migrations/v12_26_3-1632339710755-move_nlu_config_in_main_config.ts

```diff
@@ -0,0 +1,163 @@
+  queueTrainingOnBotMount: false,
+  legacyElection: false,
+  modelCacheSize: '850mb',
+  maxTrainingPerInstance: 2,
+  ducklingEnabled: true,
+  ducklingURL: 'https://duckling.botpress.io',
+  languageSources: [{ endpoint: 'https://lang-01.botpress.io' }]
+}
+
+const mapModConfigToCore = (modConfig: Partial<NLUModConfig>): Partial<NLUCoreConfig> => {
+  const {
+    ducklingEnabled,
+    ducklingURL,
+    languageSources,
+    modelCacheSize,
+    maxTrainingPerInstance,
+    queueTrainingOnBotMount,
+    legacyElection,
+    nluServer
+  } = modConfig
+
+  let autoStartNLUServer: boolean | undefined
+  let nluServerEndpoint: string | undefined
+  if (nluServer && nluServer.autoStart) {
+    autoStartNLUServer = true
+  } else if (nluServer && !nluServer.autoStart) {
+    autoStartNLUServer = false
+    nluServerEndpoint = nluServer.endpoint
+  }
+
+  return {
+    autoStartNLUServer,
+    nluServerEndpoint,
+    legacyElection,
+    queueTrainingOnBotMount,
+    maxTrainingPerInstance,
+    modelCacheSize,
+    languageSources,
+    ducklingURL,
+    ducklingEnabled
+  }
+}
+
+const mapCoreConfigToMod = (coreConfig: Partial<NLUCoreConfig>): Partial<NLUModConfig> => {
+  const {
+    ducklingEnabled,
+    ducklingURL,
+    languageSources,
+    modelCacheSize,
+    maxTrainingPerInstance,
+    queueTrainingOnBotMount,
+    legacyElection,
+    autoStartNLUServer,
+    nluServerEndpoint
+  } = coreConfig
+
+  let nluServer: { autoStart: true } | ({ autoStart: false } & LanguageSource) | undefined
+  if (autoStartNLUServer) {
+    nluServer = { autoStart: true }
+  } else if (nluServerEndpoint) {
+    nluServer = { autoStart: false, endpoint: nluServerEndpoint }
+  }
+
+  return {
+    nluServer,
+    legacyElection: legacyElection ?? false,
+    queueTrainingOnBotMount,
+    maxTrainingPerInstance,
+    modelCacheSize: modelCacheSize ?? '850mb',
+    languageSources,
+    ducklingURL,
+    ducklingEnabled
+  }
+}
+
+const migration: Migration = {
+  info: {
+    description: '',
+    target: 'core',
+    type: 'config'
+  },
+  up: async ({ bp, metadata }: MigrationOpts): Promise<sdk.MigrationResult> => {
+    try {
+      const ghost = bp.ghost.forGlobal()
+
+      const nluJsonExists = await ghost.fileExists('config', 'nlu.json')
+
+      let nluCoreConfig: NLUCoreConfig
+      if (!nluJsonExists) {
+        nluCoreConfig = DEFAULT_NLU_CORE_CONFIG
+      } else {
+        const nluModConfig = await ghost.readFileAsObject<NLUModConfig>('config', 'nlu.json')
+        await ghost.deleteFile('config', 'nlu.json')
+        nluCoreConfig = { ...DEFAULT_NLU_CORE_CONFIG, ...mapModConfigToCore(nluModConfig) }
+      }
+
+      const currentBotpressConfig = await ghost.readFileAsObject<BotpressConfig>('.', 'botpress.config.json')
+      const updatedConfig: BotpressConfig = { ...currentBotpressConfig, nlu: nluCoreConfig }
+      await ghost.upsertFile('.', 'botpress.config.json', JSON.stringify(updatedConfig, null, 2))
```

Comment:
These 3 lines can be combined: `configProvider.mergeBotpressConfig({ nlu: nluCoreConfig })`

Reply:
yeah I know but the method is not exposed through the botpress object

Reply:
You are in the core, you don't need the sdk.

![image](https://user-images.githubusercontent.com/42552874/134549899-3ac26545-400f-4111-9713-0fec74ab4f12.png)

Reply:
didnt know that :+1: